File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_020)_Maximum_Difference.cpp
#include<iostream>
using namespace std;

int maxDiff(int arr[], int n)
{
    int res = arr[1]-arr[0], minVal = arr[0];
    for(int i=1; i<n; i++)
    {
        res = max(res, arr[i]-minVal);
        minVal = min(minVal, arr[i]);
    }
    return res;
}

int main()
{
    int arr[] = {2, 3, 10, 6, 4, 8, 1}, n=7;
    cout<<"Maximum Difference = "<<maxDiff(arr,n)<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/JavaScript/_020)_Maximum_Difference.js
function maxDiff(arr) {
    let n = arr.length;
    let res = arr[1] - arr[0],
        minVal = arr[0];
    for (let i = 1; i < n; i++) {
        res = Math.max(res, arr[i] - minVal);
        minVal = Math.min(minVal, arr[i]);
    }
    return res;
}

let arr = [2, 3, 10, 6, 4, 8, 1];
console.log(`Maximum Difference = ${maxDiff(arr)}`);
